EUR-Lex.js

```diff
@@ -103,96 +135,125 @@ function doWeb(doc, url) {
+	HU: 'hun',
+	IT: 'ita',
+	LV: 'lav',
+	LT: 'lit',
+	MT: 'mlt',
+	NL: 'nld',
+	PL: 'pol',
+	PT: 'por',
+	RO: 'ron',
+	SK: 'slk',
+	SL: 'slv',
+	SV: 'swe'
 };
 
 
 function scrape(doc, url) {
 	var type = detectWeb(doc, url);
 	var item = new Zotero.Item(type);
-	
+
 	// determine the language we are currently looking the document at
-	var languageUrl = url.split('/')[4];
-	if (languageUrl=="AUTO") {
+	let languageUrl = url.split('/')[4];
+	if (languageUrl === "AUTO") {
 		languageUrl = autoLanguage || "EN";
 	}
-	var language = languageMapping[languageUrl] || "eng";
-	
+	const language = languageMapping[languageUrl] || "eng";
+
 	item.title = attr(doc, 'meta[property="eli:title"][lang=' + languageUrl.toLowerCase() + ']', 'content');
 	item.language = languageUrl.toLowerCase();
-	
+
 	var uri = attr(doc, '#format_language_table_digital_sign_act_' + languageUrl.toUpperCase(), 'href');
 	if (uri) {
-		var uriParts = uri.split('/').pop().replace('?uri=', '').split(':');
+		var uriParts = uri.split('/').pop().replace('?uri=', '')
+			.split(':');
 		// e.g. uriParts =  ["OJ", "L", "1995", "281", "TOC"]
 		// e.g. uriParts = ["DD", "03", "061", "TOC", "FI"]
-		if (uriParts.length>=4) {
+		if (uriParts.length >= 4) {
 			if (/\d+/.test(uriParts[1])) {
 				item.code = uriParts[0];
 				item.codeNumber = uriParts[1] + ', ' + uriParts[2];
-			} else {
+			}
+			else {
 				item.code = uriParts[0] + ' ' + uriParts[1];
 				item.codeNumber = uriParts[3];
 			}
-			if (type=="bill") {
+			if (type == "bill") {
 				item.codeVolume = item.code;
 				item.code = item.codeNumber;
 			}
 		}
 	}
-	
+
 	item.number = attr(doc, 'meta[property="eli:id_local"]', 'content');
-	
+
 	item.date = attr(doc, 'meta[property="eli:date_publication"]', 'content');
 	// attr(doc, 'meta[property="eli:date_document"]', 'content');
 
 	var passedBy = doc.querySelectorAll('meta[property="eli:passed_by"]');
 	var passedByArray = [];
-	for (let i=0; i<passedBy.length; i++) {
+	for (let i = 0; i < passedBy.length; i++) {
 		passedByArray.push(passedBy[i].getAttribute('resource').split('/').pop());
 	}
 	item.legislativeBody = passedByArray.join(', ');
-	
+
 	item.url = attr(doc, 'meta[typeOf="eli:LegalResource"]', 'about') + '/' + language;
-	
+
 	// eli:is_about -> eurovoc -> tags
-	
+
+	item.complete();
+}
+
+function scrapeCase(doc, url) {
+	const type = detectWeb(doc, url);
+	const item = new Zotero.Item(type);
+
+	// determine the language we are currently looking the document at
+	let languageUrl = url.split('/')[4];
+	if (languageUrl == "AUTO") {
+		languageUrl = autoLanguage || "EN";
+	}
+	// const language = languageMapping[languageUrl] || "eng";
+
+	item.title = ZU.xpathText(doc, "//p[@id='translatedTitle']").split('.')[1].trim();
+	item.language = languageUrl.toLowerCase();
+
+
+	const parsedDate = ZU.xpathText(doc, "//div[@id='PPDates_Contents']/div[1]/dl[1]/dd[1]").split('/');
+	item.date = parsedDate[2] + '-' + parsedDate[1] + '-' + parsedDate[0];
+
+
+	item.url = url;
```

Comment:
As far as a I see the only difference for scraping cases is the part about the date. But we can this then also integrate above in the normal scrape function with an if clause. There is IMO no need for a different function `scrapeCase`.